src/OpenTelemetry.Collector.Azure/AzureSdkCollector.cs

```diff
@@ -0,0 +1,150 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+namespace OpenTelemetry.Collectors.Azure
+{
+    using System;
+    using System.Collections.Concurrent;
+    using System.Collections.Generic;
+    using System.Diagnostics;
+    using OpenTelemetry.Context;
+    using OpenTelemetry.Trace;
+
+    public class AzureSdkCollector : IDisposable, IObserver<DiagnosticListener>, IObserver<KeyValuePair<string, object>>
+    {
+        private readonly ConcurrentDictionary<Activity, IScope> scopes = new ConcurrentDictionary<Activity, IScope>(new ActivityReferenceEqualityComparer());
+
+        private readonly ITracer tracer;
+
+        private readonly ISampler sampler;
+
+        private List<IDisposable> subscriptions = new List<IDisposable>();
+
+        public AzureSdkCollector(ITracer tracer, ISampler sampler)
+        {
+            this.tracer = tracer;
+            this.sampler = sampler;
+
+            this.subscriptions.Add(DiagnosticListener.AllListeners.Subscribe(this));
+        }
+
+        public void Dispose()
+        {
+            lock (this.subscriptions)
+            {
+                foreach (var subscription in this.subscriptions)
+                {
+                    subscription.Dispose();
+                }
+            }
+        }
+
+        public void OnCompleted()
+        {
+        }
+
+        public void OnError(Exception error)
+        {
+        }
+
+        public void OnNext(KeyValuePair<string, object> value)
+        {
+            if (value.Key.EndsWith("Start"))
+            {
+                this.OnStartActivity(Activity.Current, value.Value);
+            }
+            else if (value.Key.EndsWith("Stop"))
+            {
+                // Current.Parent is used because OT wraps additional Activity over
+                this.OnStopActivity(Activity.Current.Parent, value.Value);
+            }
+            else if (value.Key.EndsWith("Exception"))
+            {
+                // Current.Parent is used because OT wraps additional Activity over
+                this.OnException(Activity.Current.Parent, value.Value);
+            }
+        }
+
+        public void OnNext(DiagnosticListener value)
+        {
+            if (value.Name.StartsWith("Azure"))
+            {
+                lock (this.subscriptions)
+                {
+                    this.subscriptions.Add(value.Subscribe(this));
+                }
+            }
+        }
+
+        private void OnStartActivity(Activity current, object valueValue)
+        {
+            var operationName = current.OperationName;
+            foreach (var keyValuePair in current.Tags)
+            {
+                if (keyValuePair.Key == "http.url")
+                {
+                    operationName = keyValuePair.Value;
+                }
+            }
+
+            var span = this.tracer.SpanBuilder(operationName)
+                .SetSampler(this.sampler)
```

Comment:
add `SetCreateChild(false)` here -it'll tell OT to NOT create another Activity

Reply:
also set kind - Client

Reply:
Now HTTP Client calls are not attached to the correct spans.
![image](https://user-images.githubusercontent.com/1697911/61968650-6a4a1400-af8d-11e9-8d85-85bcab20b74a.png)

Reply:
By Httpclient calls you mean auto-tracked or Azure SDK http policy? 
auto-tracked (double-tracked) go away with https://github.com/open-telemetry/opentelemetry-dotnet/pull/162.

Can you give me a repro or let's have a quick team call to debug it?

Reply:
Let's wait for your changes to go in and I'll test again.

Reply:
they're in, please rebase

Reply:
Zipkin gets flooded by span upload requests (https://github.com/open-telemetry/opentelemetry-dotnet/issues/164) and seems like it never exports the original trace or I'm not able to find it.

Reply:
I've sent temp fix for #164 - #167. Do you still have a problem with missing spans or you fixed it with the attribute value.ToString() in Zipkin Exporter?

Is it still relevant?

Reply:
Looks great!
![image](https://user-images.githubusercontent.com/1697911/62155722-79e39880-b2be-11e9-8948-7586dab30366.png)